src/azul/queues.py

```diff
@@ -165,30 +166,51 @@ def count_messages(cls, queues: Mapping[str, Any]) -> int:
         return total_message_count
 
     @classmethod
-    def wait_for_queue_level(cls, empty: bool = True, num_bundles: int = None):
+    def wait_for_queue_level(cls, empty: bool = True,
+                             num_bundles: int = None,
+                             min_timeout: Optional[int] = None,
+                             max_timeout: Optional[int] = None):
         """
         Wait until the total count of messages in the notify and document queues
         reaches the desired level
 
+        :param min_timeout: Specifies minimum wait time for queues to process
+                            index notifications. If notifications with a prefix
+                            are being process the recommended value for this
+                            field is 10 minutes.
+        :param max_timeout: Specifies maximum wait time for queues to process
+                            index notifications.
         :param empty: True to wait until the queues are empty, False to wait until not empty.
         :param num_bundles: Number of bundles being indexed (None = many bundles)
         """
+
+        def _compute_timeout(num_bundles):
+            # It takes approx. 6 seconds per worker to process a bundle.
+            timeout = 6 * num_bundles / config.indexer_concurrency
+            if max_timeout is not None:
+                timeout = min(timeout, max_timeout)
+                logger.info('Maximum wait time is %i seconds', timeout)
+            elif min_timeout is not None or config.dss_query_prefix:
+                timeout = max(timeout, min_timeout if not config.dss_query_prefix else 60 * 10)
+                logger.info('Minimum wait time is %i seconds', timeout)
+            return timeout
+
         sleep_time = 5
         deque_size = 10 if empty else 1
         queues = cls.get_queues(config.work_queue_names)
         queue_size_history = deque(maxlen=deque_size)
         wait_start_time = time.time()
+        compute_timeout = None
```

Comment:
This variable is the core of the dynamic calculation of the timeout inside the while loop. Checkout how is used and if it is an elegant approach.

Reply:
Not sure, yet.

Reply:
Don't like it. This solution requires two identifiers: `compute_timeout` and `_compute_timeout`. So would a solution with a boolean flag and a function but that implementation is less confusing.